### PR TITLE
Add fixed and sortable widgets

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
*Issue #:*
https://github.com/bcgov/met-public/issues/885

*Description of changes:*
Makes engagement phases widget fixed at the top.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/engagement/form/EngagementWidgets/WidgetsBlock.tsx [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
     const { savedEngagement } = useContext(ActionContext);
     const dispatch = useAppDispatch();
 
-    const [tempWidgets, setTempWidgets] = useState<Widget[]>(widgets);
+    const [sortableWidgets, setSortableWidgets] = useState<Widget[]>([]);
+    const [fixedWidgets, setFixedWidgets] = useState<Widget[]>([]);
 
     useEffect(() => {
-        setTempWidgets(widgets);
+        const widgetsFixedList = widgets.filter((w) => w.widget_type_id == WidgetType.Phases);
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/form/EngagementWidgets/WidgetsBlock.tsx:25]: Maybe we can just define an array of sortable types [WidgetType.Phases, ..] and just check if type id is in it. This way we can just extend the array if mor fixed types existed

Comment (tiagograf-AOT) [reply met-web/src/components/engagement/form/EngagementWidgets/WidgetsBlock.tsx]: Done

File: met-web/src/components/engagement/form/EngagementWidgets/WidgetsBlock.tsx [hunk 2]
```diff
@@ -33,18 +39,19 @@ const WidgetsBlock = () => {
         handleWidgetDrawerOpen(true);
     };
 
-    const moveWidget = useCallback((dragIndex: number, hoverIndex: number) => {
-        setTempWidgets((prevWidgets: Widget[]) => {
+    const moveWidget = (dragIndex: number, hoverIndex: number) => {
```
Comment (tiagograf-AOT) [inline met-web/src/components/engagement/form/EngagementWidgets/WidgetsBlock.tsx:42]: Not sure what was the reason to use a callback function here, but I had to remove it to have access to fixedWidgets

**Review by djnunez-aot — APPROVED** [review]

**Review by jadmsaadaot — APPROVED** [review]
Approved, just a bit concerned that without useCallback there might be more calls happening. A refactor for later might be in the backend where simplify the validation. Maybe make sort_index nullable.

**Pull request merged**